admin-dev/themes/new-theme/js/vue/components/Pagination.vue

```diff
@@ -0,0 +1,159 @@
+<!--**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ *-->
+<template>
+  <div class="pagination">
+    <ul class="pagination-list">
+      <li
+        class="pagination-item pagination-previous"
+        @click="goToPage(currentPage - 1)"
+      >
+        <i class="material-icons">chevron_left</i>
+      </li>
+      <li
+        :class="['pagination-item', isActive(key)]"
+        v-for="(page, key) of paginatedDatas"
+        :key="key"
+        @click="goToPage(key + 1)"
+      >
+        {{ key + 1 }}
+      </li>
+      <li
+        class="pagination-item pagination-next"
+        @click="goToPage(currentPage + 1)"
+      >
+        <i class="material-icons">chevron_right</i>
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+  export default {
+    name: 'Pagination',
+    data() {
+      return {
+        paginatedDatas: [],
+        currentPage: 1,
+      };
+    },
+    props: {
+      datas: {
+        type: Array,
+        default: () => [],
+      },
+      paginationLength: {
+        type: Number,
+        default: 14,
```

Comment:
Why 14 ? Can't we have a const here ?

Reply:
Where would you want me to add a constant? We never did it yet on default value of vue components props

14 is the number of items shown by pages (by the pagination) by default, but you can setup this when you use the component by passing paginationLength as a prop